src/Detector/PlatformDetectorProvider.cs

```diff
@@ -0,0 +1,61 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Logging;
+using Microsoft.Oryx.Detector.DotNetCore;
+using Microsoft.Oryx.Detector.Node;
+using Microsoft.Oryx.Detector.Php;
+using Microsoft.Oryx.Detector.Python;
+
+
+namespace Microsoft.Oryx.Detector
+
+{
+    public class PlatformDetectorProvider : IPlatformDetectorProvider
```

Comment:
You do not need this api since one can register a list of detectors in DI

Reply:
I think it will be easier to extend if we have it, such as adding a hugo detector in the future. We iterate through all detectors, we don't need to add each detectors but call this api in `DefaultPlatformDetector.cs`

Reply:
> we don't need to add each detectors but call this api in DefaultPlatformDetector.cs

Let's say if you write a new detector, we still would have to add it to DI and also over here which feels unnecessary. You can take a look at the following api in current build script generator

```
services.TryAddEnumerable(
                ServiceDescriptor.Singleton<IProgrammingPlatform, NodePlatform>());
```

We can add detectors similar to the above and the `DefaultPlatformDetector` could inject in `IEnumerable<IPlatformDetector>` and loop through them.

Reply:
Got it, that's a good point.. I'll try to use this way instead